api/ws-server.js

```diff
@@ -20,7 +20,14 @@ const setupWss = async (server, sessionStore) => {
   io.on('connection', (client) => {
     logger.info(`Web client connected: ID: ${client.id}`)
     const user = { ...client.request.user.dataValues }
-    clients[user.id] = { socketID: client.id, user }
+
+    if (clients[user.id] && clients[user.id].sockets) {
+      clients[user.id].sockets.push(client.id)
+    } else {
+      clients[user.id] = { sockets: [client.id], user }
+    }
```

Comment:
Better written as:

```
if (!clients[user.id]) {
  clients[user.id] = { sockets: [], user }
}
clients[user.id].sockets.push(client.id)
```

Reply:
Are we certain that if `clients[user.id]` exists, then also `.sockets` exists?

Reply:
Yes – the entry is only created at this point and `sockets` is never removed from it.